compiler/testData/ir/irText/expressions/sam/samConversionToGeneric.txt

```diff
@@ -50,7 +50,7 @@ FILE fqName:<root> fileName:/samConversionToGeneric.kt
       CALL 'public open fun bar <X> (j: @[FlexibleNullability] <root>.J<@[FlexibleNullability] X of <root>.H.bar?>?): kotlin.Unit declared in <root>.H' type=kotlin.Unit origin=null
         <X>: @[FlexibleNullability] kotlin.String?
         j: TYPE_OP type=@[FlexibleNullability] <root>.J<@[FlexibleNullability] kotlin.String?>? origin=SAM_CONVERSION typeOperand=@[FlexibleNullability] <root>.J<@[FlexibleNullability] kotlin.String?>?
-          TYPE_OP type=kotlin.Function1<@[ParameterName(name = 'x')] @[FlexibleNullability] kotlin.String?, @[FlexibleNullability] kotlin.String?> origin=IMPLICIT_CAST typeOperand=kotlin.Function1<@[ParameterName(name = 'x')] @[FlexibleNullability] kotlin.String?, @[FlexibleNullability] kotlin.String?>
+          TYPE_OP type=kotlin.Function1<kotlin.String, kotlin.String> origin=IMPLICIT_CAST typeOperand=kotlin.Function1<kotlin.String, kotlin.String>
```

Comment:
Minor: looks like we are losing some type information here (although it doesn't look dangerous in case of SAM conversions).

Reply:
Please, add a box test where flexible nullability of value parameters of the corresponding lambda would matter.
Something like:
```
// FILE: test.kt
fun test(a: Any) {
    a as (String) -> String
    J.use(a)
}

fun box(): String {
    test({s: String? -> s})
    return "OK"
}

// FILE: JFoo.java
public interface JFoo<T> {
    T foo(T x);
}

// FILE: J.java
public class J {
    public static void use(JFoo<String> jfoo) {
        jfoo.foo(null);
    }
}
```

Reply:
Done.